crates/contracts/solidity/PhonyERC20.sol

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.16;
+
+import { Caller } from "./libraries/Caller.sol";
+import { SafeERC20 } from "./libraries/SafeERC20.sol";
+
+/// @dev A phony ERC20 implementation, that replaces the bytecode of an existing
+/// on-chain contract and allows minting to arbitrary addresses. This can be
+/// used to fund an account when one with a known balance can't be found.
+contract PhonyERC20 {
+    using Caller for *;
+    using SafeERC20 for *;
+
+    /// @dev A seed to offset all internal balance storage operations to make
+    /// sure that we don't accidentally touch any of the implementation's slots.
+    /// Derived from:
+    /// ```
+    /// keccak("hakuna matata") - 1
```

Comment:
Why `- 1`?

Reply:
To avoid collisions. So, if you know how the pre-image of a digest, you could conceivably have a collision by carefully crafting some input.

One strategy is to hash a known pre-image, then subtract or add 1. Now the digest has a documented way of creating it (this means I didn't pick a digest with some special properties that I would be able to abuse) but has no known pre-image, making collisions impossible.

I can add a short comment.

Reply:
Makes sense. If it is a common strategy then no need to document.